File: ios/TsMediapipeViewManager.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE (TsMediapipeViewManager, RCTViewManager)

// Props envoyées à la View
RCT_EXPORT_VIEW_PROPERTY(width, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(height, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(onLandmark, RCTDirectEventBlock)

RCT_EXPORT_VIEW_PROPERTY(face, BOOL)
RCT_EXPORT_VIEW_PROPERTY(leftArm, BOOL)
RCT_EXPORT_VIEW_PROPERTY(rightArm, BOOL)
RCT_EXPORT_VIEW_PROPERTY(leftWrist, BOOL)
RCT_EXPORT_VIEW_PROPERTY(rightWrist, BOOL)
RCT_EXPORT_VIEW_PROPERTY(torso, BOOL)
RCT_EXPORT_VIEW_PROPERTY(leftLeg, BOOL)
RCT_EXPORT_VIEW_PROPERTY(rightLeg, BOOL)
RCT_EXPORT_VIEW_PROPERTY(leftAnkle, BOOL)
RCT_EXPORT_VIEW_PROPERTY(rightAnkle, BOOL)
RCT_EXPORT_VIEW_PROPERTY(frameLimit, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(activeLandmarks, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(activeLandmarkRadius, NSNumber)

// Méthodes natives
RCT_EXTERN_METHOD(switchCamera)

RCT_EXTERN_METHOD(startRecording : (RCTPromiseResolveBlock)
                      resolve rejecter : (RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(stopRecording : (RCTPromiseResolveBlock)
                      resolve rejecter : (RCTPromiseRejectBlock)reject)

@end

File: ios/TsMediapipeViewManager.swift
import Foundation
import React

@objc(TsMediapipeViewManager)
class TsMediapipeViewManager: RCTViewManager {

  private var cameraView: CameraView?

  override func view() -> (UIView) {
    let view = CameraView()
    cameraView = view
    return view
  }

  @objc override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc func switchCamera() {
    cameraView?.switchCamera()
  }

  @objc func startRecording(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {

    DispatchQueue.main.async {
      guard let cameraView = self.cameraView else {
        reject("NO_VIEW", "CameraView not initialized", nil)
        return
      }

      do {
        try cameraView.startRecording()
        resolve(nil)
      } catch {
        reject("ERR", error.localizedDescription, error)
      }
    }
  }

  @objc func stopRecording(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {

    DispatchQueue.main.async {
      guard let cameraView = self.cameraView else {
        reject("NO_VIEW", "CameraView not initialized", nil)
        return
      }

      cameraView.stopRecording { path in
        resolve(path)
      }
    }
  }
}
